manifests/istio-control/istio-autoinject/files/injection-template.yaml

```diff
@@ -319,12 +319,16 @@ template: |
         add:
         - NET_ADMIN
         {{- end }}
+        {{ if eq (annotation .ObjectMeta `sidecar.istio.io/interceptionMode` .ProxyConfig.InterceptionMode) `NONE` -}}
+        add:
+        - CAP_NET_BIND_SERVICE
+        {{- end }}
         drop:
         - ALL
       privileged: {{ .Values.global.proxy.privileged }}
       readOnlyRootFilesystem: {{ not .Values.global.proxy.enableCoreDump }}
       runAsGroup: 1337
-      {{ if eq (annotation .ObjectMeta `sidecar.istio.io/interceptionMode` .ProxyConfig.InterceptionMode) `TPROXY` -}}
+      {{ if or (eq (annotation .ObjectMeta `sidecar.istio.io/interceptionMode` .ProxyConfig.InterceptionMode) `TPROXY`) (eq (annotation .ObjectMeta `sidecar.istio.io/interceptionMode` .ProxyConfig.InterceptionMode) `NONE`) -}}
```

Comment:
Why do you need root when you already obtain CAP_NET_BIND_SERVICE?

Reply:
+1
This shouldn't be needed. I took a quick look at the kernel code to confirm that.

However, I've remarked that `CAP_NET_RAW` doesn't work without UID 0 in Docker containers, and I don't understand why. Maybe it's the same with `CAP_NET_BIND_SERVICE`?
@jackkleeman can you explain why this doesn't work without UID 0?

Reply:
https://github.com/moby/moby/issues/8460 somewhat unfortunate. if you google you'll find many simiilar issues

Reply:
>However, I've remarked that CAP_NET_RAW doesn't work without UID 0 in Docker containers, and I don't understand why

yep, same thing

Reply:
Ah it's still the ambient capabilities bug in Docker that has still not been fixed in years. Got it. 😞